### ISSUE TITLE
Make it easy to generate an ISO with default NixOS configuration

### ISSUE THREAD

**Issue by bouk** [NixOS/calamares-nixos-extensions#15]
Hi, I want to customize the GUI NixOS installation CD by adding a bunch of configuration that should be set by default (e.g. enabling sshd, adding ssh keys). It would be cool if it was easier to customize the set-up process to include default configuration.

**Comment by IvarWithoutBones** [comment]
Having the ability to add snippets to the generated `configuration.nix` would be pretty useful for me as well. I was wondering if there is a clear path to move forward with this issue?

 A solution that comes to mind would be to simply hardcode in a string at build time (a plain `substituteAll` would do the trick), which could be exposed to the end user through overrides. Not sure if that is the best  way of going about this though, so I figured I'd ask for some feedback :)

**Comment by vlinkz** [comment]
Yes, I think something like that would work. Have a seperate `configuration.nix` file with some placeholders that the installer will replace with the user options. I actually recently did something similar with the [Icicle installer](https://github.com/snowflakelinux/icicle/tree/main/config)

**Comment by ElvishJerricco** [comment]
This repo has been merged upstream into nixpkgs and will be archived. If this is still an issue, please open a new one there.
